app/main.py

```diff
@@ -1,3 +1,67 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(
+                km=self.km + other
+            )
+        else:
+            return Distance(
+                km=self.km + other.km
+            )
+
+    def __iadd__(self, other: Distance | float) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
```

Comment:
You can try to omit condition duplicating here, with implementing `__add__` and swapping the parameter in __iadd__, calling the __add__ method, a common practice

Reply:
sure, thank you, changed